airflow/models/taskinstance.py

```diff
@@ -2966,12 +2966,9 @@ def as_dict(self):
             stacklevel=2,
         )
         new_dict = dict(self.__dict__)
-        for key in new_dict:
-            if key in ["start_date", "end_date"]:
-                val = new_dict[key]
-                if not val or isinstance(val, str):
-                    continue
-                new_dict.update({key: val.isoformat()})
+        for key, val in new_dict.items():
+            if key in ["start_date", "end_date"] and isinstance(val, datetime):
```

Comment:
I think this is not exactly the same. We can have pendulum datetime here as well.